include/aikido/control/ros/RosTrajectoryExecutor.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef AIKIDO_CONTROL_ROS_ROSTRAJECTORYEXECUTOR_HPP_
+#define AIKIDO_CONTROL_ROS_ROSTRAJECTORYEXECUTOR_HPP_
+#include <chrono>
+#include <future>
+#include <mutex>
+#include <ros/ros.h>
+#include <ros/callback_queue.h>
+#include <control_msgs/FollowJointTrajectoryAction.h>
+#include <dart/dart.hpp>
+#include <aikido/control/TrajectoryExecutor.hpp>
+#include <aikido/trajectory/Trajectory.hpp>
+
+// actionlib and DART both #define this macro.
+#undef DEPRECATED
+#include <actionlib/client/action_client.h>
+#undef DEPRECATED
+
+namespace aikido {
+namespace control {
+namespace ros {
+
+class RosTrajectoryExecutor : public aikido::control::TrajectoryExecutor
+{
+public:
+  RosTrajectoryExecutor(
+    ::dart::dynamics::MetaSkeletonPtr skeleton, 
+    ::ros::NodeHandle node,
+    const std::string& serverName,
+    double timestep,
+    double goalTimeTolerance,
+    std::chrono::milliseconds connectionTimeout
+      = std::chrono::milliseconds{1000},
+    std::chrono::milliseconds connectionPollingRate
+      = std::chrono::milliseconds{20}
+  );
+
+  virtual ~RosTrajectoryExecutor();
+
+  std::future<void> execute(trajectory::TrajectoryPtr _traj) override;
+
+  std::future<void> execute(
+    trajectory::TrajectoryPtr _traj, const ::ros::Time& _startTime);
+
+  /// Simulates mTraj. To be executed on a separate thread.
+  void spin();
```

Comment:
We should refactor this function into a `spinOnce`-style callback that we can call from the `Executor` implemented in #151. That is potentially important because it lets us consolidate all of our various callbacks, e.g. multiple `RosTrajectoryExecutor`s and `JointStateSubscriber`s, into a single thread.